src/shared/components/post/post-form.tsx

```diff
@@ -51,12 +50,13 @@ const MAX_POST_TITLE_LENGTH = 200;
 
 interface PostFormProps {
   post_view?: PostView; // If a post is given, that means this is an edit
+  crossPosts?: PostView[];
   allLanguages: Language[];
   siteLanguages: number[];
   params?: PostFormParams;
-  onCancel?(): any;
-  onCreate?(post: PostView): any;
-  onEdit?(post: PostView): any;
+  onCancel?(): void;
+  onCreate?(form: CreatePost): Promise<RequestState<PostResponse>>;
```

Comment:
Same stuff here. Only have this pass up the form.

Check the `CreateCommunity` page and `CommunityForm` for an example very close to this.

Reply:
Resolved from changes merged from `use_http_client_dess`.